pinot-common/src/main/java/com/linkedin/pinot/common/utils/primitive/Bytes.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.utils.primitive;
+
+import java.util.Arrays;
+import javax.annotation.Nonnull;
+import org.apache.commons.codec.binary.Hex;
+
+
+/**
+ * Wrapper around byte[] that provides additional features such as:
+ * <ul>
+ *   <li> Implements comparable interface, so comparison and sorting can be performed. </li>
+ *   <li> Implements equals() and hashCode(), so it can be used as key for HashMap/Set. </li>
+ * </ul>
+ */
+public class Bytes implements Comparable<Bytes> {
```

Comment:
Would make more sense to name this `ByteArray`, and it would avoid the name clash with Guava `Bytes`.

Reply:
+1 on calling it ByteArray

Reply:
Yeah, I used Bytes to be compatible with BYTES type in the schema. But will change to ByteArray.